openapi/openapi-v2.json

```diff
@@ -422,6 +425,602 @@
                     }
                 }
             }
+        },
+        "/questionnaires/{questionnaireId}/progress-entries": {
+            "get": {
+                "tags": ["ProgressEntries"],
+                "summary": "Get progress entries for a questionnaire.",
+                "description": "Gets all progress entries for a questionnaire. The results can be filtered.",
+                "operationId": "getProgressEntries",
+                "parameters": [
+                    {
+                        "name": "questionnaireId",
+                        "in": "path",
+                        "required": true,
+                        "description": "The id of the specific questionnaire instance. Format UUIDv4.",
+                        "schema": {
+                            "title": "UUID v4",
+                            "type": "string",
+                            "pattern": "^[0-9a-f]{8}-[0-9a-f]{4}-4[0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$"
+                        },
+                        "example": "285cb104-0c15-4a9c-9840-cb1007f098fb"
+                    },
+                    {
+                        "in": "query",
+                        "name": "filter",
+                        "description": "Filter progress entries",
+                        "schema": {
+                            "type": "object",
+                            "additionalProperties": false,
+                            "properties": {
+                                "position": {
+                                    "enum": ["current", "first"]
+                                },
+                                "sectionId": {
+                                    "title": "Section Id",
+                                    "type": "string",
+                                    "pattern": "^(?:p-?(?:-[a-z0-9]{1,20}){1,20}|system|referrer)$"
+                                }
```

Comment:
are we removing `referrer` from `sectionId`?

Reply:
Is there a test to check that system/referrer/owner are valid, and any other word is not valid

Reply:
We've got a test that says it should 404 if it doesn't match that regex, but not specifically testing those 3 values